engine/src/main/java/com/datatorrent/stram/plan/logical/LogicalPlan.java

```diff
@@ -1374,7 +1425,8 @@ private void addDAGToCurrentDAG(ModuleMeta moduleMeta)
       InputPort[] inputPorts = ports.toArray(new InputPort[]{});
 
       name = subDAGName + MODULE_NAMESPACE_SEPARATOR + streamMeta.getName();
-      this.addStream(name, sourceMeta.getPortObject(), inputPorts);
+      StreamMeta streamMetaNew = this.addStream(name, sourceMeta.getPortObject(), inputPorts);
+      streamMetaNew.copyFrom(streamMeta);
```

Comment:
instead of copyFrom can we do
`streamMetaNew.setLocality(streamMeta.getLocality());`?